store.rb

```diff
@@ -26,23 +55,47 @@ def start()
     end
     @zk.ping?
     @log.info "ZK connection established successfully. session_id=#{session_id}"
+
+    if @cache_enabled then
+      watch
+      reload_instances
+      start_fetch_thread
+    end
   end
 
   def session_id
     '0x%x' % @zk.session_id rescue nil
   end
 
+  def stop_cache_related()
+    @cache_root_watcher.unsubscribe if @cache_root_watcher
+    @cache_root_watcher = nil
+    @cache_fetch_thread_should_run = false
```

Comment:
non-obviously, because you only set this to `true` in the initializer, the thread will now no longer run if you call `start` after calling `stop`; i guess you do need to set it to true in `start_fetch_thread`